pytest_timeout.py

```diff
@@ -414,7 +423,14 @@ def timeout_timer(item, timeout):
     finally:
         sys.stdout.flush()
         sys.stderr.flush()
-        os._exit(1)
+
+    for _ in range(10):
+        ctypes.pythonapi.PyThreadState_SetAsyncExc(
+            ctypes.c_long(thread.ident), ctypes.py_object(Failed),
+        )
+        time.sleep(0.5)
+
+    os._exit(1)
```

Comment:
Moved out of finally block. The intention of this is to try to avoid crashing the worker, if possible. If the worker is still going after 5 seconds, we failed (timeout thread is not cancelled), so we crash the worker.

Reply:
I get the delay, but why does it need to set the exception more than once?